addons/mail/models/mail_channel.py

```diff
@@ -1153,6 +1158,28 @@ def channel_create(self, name, privacy='public'):
         self.env['bus.bus'].sendone((self._cr.dbname, 'res.partner', self.env.user.partner_id.id), channel_info)
         return channel_info
 
+    @api.model
+    def group_chat_create(self, partners_to):
+        """ Create a private group chat with at least 3 partners.
+            :param partners_to : list of res.partner ids to add to the conversation
+            :returns: channel_info of the created or existing channel
+            :rtype: dict
+        """
+        if self.env.user.partner_id.id not in partners_to:
+            partners_to.append(self.env.user.partner_id.id)
+        channel = self.create({
+            'channel_partner_ids': [Command.link(partner_id) for partner_id in partners_to],
+            'public': 'private',
+            'channel_type': 'group',
+            'email_send': False,
+            'name': ', '.join(self.env['res.partner'].sudo().browse(partners_to).mapped('name')),
+        })
+        self.env['mail.channel.partner'].search(
+            [('partner_id', '=', self.env.user.partner_id.id), ('channel_id', '=', channel.id)]).write(
+            {'is_pinned': True})
```

Comment:
I think it should be pinned for all users, not only the current user.
Then the search would be equivalent to simply accessing `channel.channel_last_seen_partner_ids`